keras/models.py

```diff
@@ -41,6 +41,18 @@ def slice_X(X, start=None, stop=None):
         else:
             return X[start:stop]
 
+def calculate_class_weights(Y, class_weight):
+    if isinstance(class_weight, dict):
+        if Y.shape[1] > 1:
+            y_classes = Y.argmax(axis=1)
+        elif Y.shape[1] == 1:
+            y_classes = np.reshape(Y, Y.shape[0])
+        else:
+            y_classes = Y
+        w = np.array(map(lambda x: class_weight[x], y_classes))
+    else:
+        w = np.ones((Y.shape[0]))
```

Comment:
Why not return `None` instead? Wouldn't it be more efficient (one less element-wise multiplication in the objective function).